### PR TITLE
Search for Project Collection Groups with Group datasource

### PR DESCRIPTION
**Pull request by tmeckel**
## All Submissions:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [x] I have updated the documentation accordingly.
* [x] I have added tests to cover my changes.
* [x] All new and existing tests passed.
* [x] My code follows the code style of this project.
* [x] I ran lint checks locally prior to submission.
* [x] Have you checked to ensure there aren't other open PRs for the same update/change?

## What about the current behavior has changed?

The PR contains a change to the Group datasource , so that it's possible to search for project collection groups and not only for groups located inside a project.

## Does this introduce a change to `go.mod`, `go.sum` or `vendor/`?

- [ ] Yes
- [x] No

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

## Any relevant logs, error output, etc?

(If it’s long, please paste to https://ghostbin.com/ and insert the link here.)

### CODE REVIEW

File: website/docs/d/data_group.html.markdown [hunk 1]
```diff
@@ -28,14 +28,27 @@ output "group_id" {
...
+
 ```
 
 ## Argument Reference
 
 The following arguments are supported:
 
-- `project_id` - (Required) The Project Id.
 - `name` - (Required) The Group Name.
+- `project_id` - (Optional) The Project Id. If no project Id is specified the project collection groups will be searched.
```
Comment (xuzhang3) [inline website/docs/d/data_group.html.markdown]: I would suggest change `Id` to `ID`

File: azuredevops/internal/service/graph/data_group.go [hunk 2]
```diff
@@ -19,15 +19,14 @@ func DataGroup() *schema.Resource {
...
-				ValidateFunc: validation.NoZeroValues,
+				ValidateFunc: validation.StringIsNotWhiteSpace,
 			},
 			"project_id": {
 				Type:         schema.TypeString,
-				ForceNew:     true,
-				Required:     true,
-				ValidateFunc: validation.NoZeroValues,
+				Optional:     true,
+				Computed:     true,
```
Comment (xuzhang3) [inline azuredevops/internal/service/graph/data_group.go]: `project_id` is required when we need to read project groups.  Both of  project groups and collection groups won't set `project_id` when service response returned. I think  we can remove `Computed:     true,` as this configuration is configured by the users.

**Review by xuzhang3 — CHANGES_REQUESTED** [review]
Few changes requested but otherwise looks good

File: website/docs/d/data_group.html.markdown (hunk 1 above)
Comment (tmeckel) [reply website/docs/d/data_group.html.markdown]: Done

File: azuredevops/internal/service/graph/data_group.go (hunk 2 above)
Comment (tmeckel) [reply azuredevops/internal/service/graph/data_group.go]: Yes I agree, you're right. I removed the `computed` flag for `project_id`

**Comment by xuzhang3** [comment]
LGTM

**Pull request merged**